src/Api/Response/Response.php

```diff
@@ -90,4 +101,104 @@ public function getResultsCount()
     }
-}
\ No newline at end of file
+
+    public function getQueryInfoMessage(): string
+    {
+        $dataQueryInfoMessage = $this->getData(self::DATA_QUERY_INFO_MESSAGE);
+
+        if (empty($dataQueryInfoMessage)) {
+            return '';
+        }
+
+        $type = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? 'did-you-mean' : $dataQueryInfoMessage['queryStringType'];
+        $alternativeQuery = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? $dataQueryInfoMessage['didYouMeanQuery']
+            : $dataQueryInfoMessage['currentQuery'];
+        $originalQuery = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? $dataQueryInfoMessage['currentQuery']
+            : $dataQueryInfoMessage['originalQuery'];
+
+        if ($type === 'corrected') {
+            return $this->translator->trans(
+                'Findologic::Template.correctedQuery',
+                [
+                    'originalQuery' => $originalQuery,
+                    'alternativeQuery' => $alternativeQuery
+                ]
+            );
+        } else if ($type === 'improved') {
+            return $this->translator->trans(
+                'Findologic::Template.improvedQuery',
+                [
+                    'originalQuery' => $originalQuery,
+                    'alternativeQuery' => $alternativeQuery
+                ]
+            );
+        } else if ($dataQueryInfoMessage['currentQuery']
+            && !empty($dataQueryInfoMessage['currentQuery'])) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageQuery',
+                [
+                    'query' => $dataQueryInfoMessage['currentQuery'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else if ($dataQueryInfoMessage['selectedCategoryName']) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageCat',
+                [
+                    'filterName' => $this->getCategoryFilterName(),
+                    'cat' => $dataQueryInfoMessage['selectedCategoryName'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else if ($dataQueryInfoMessage['selectedVendorName']) {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageVendor',
+                [
+                    'filterName' => $this->getVendorFilterName(),
+                    'vendor' => $dataQueryInfoMessage['selectedVendorName'],
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        } else {
+            return $this->translator->trans(
+                'Findologic::Template.queryInfoMessageDefault',
+                [
+                    'hits' => $this->getResultsCount()
+                ]
+            );
+        }
+    }
+
+    private function getCategoryFilterName(): string
```

Comment:
To be honest I'd rather have `null` returned than an empty string. You can drop the return type.
Meaning `null` is returned if the category filter name can not be found.